### PR TITLE
rgw: delete object in error path

### PR DESCRIPTION
**Pull request by yehudasa**
Fixes: http://tracker.ceph.com/issues/20620

Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>

### CODE REVIEW

File: src/rgw/rgw_rest_conn.cc [hunk 1]
```diff
@@ -114,8 +114,14 @@ int RGWRESTConn::put_obj_init(const rgw_user& uid, rgw_obj& obj, uint64_t obj_si
 
   param_vec_t params;
   populate_params(params, &uid, self_zone_group);
-  *req = new RGWRESTStreamWriteRequest(cct, url, NULL, &params);
-  return (*req)->put_obj_init(key, obj, obj_size, attrs);
+  RGWRESTStreamWriteRequest *wr = new RGWRESTStreamWriteRequest(cct, url, NULL, &params);
+  ret = (*req)->put_obj_init(key, obj, obj_size, attrs);
```
Comment (cbodley) [inline src/rgw/rgw_rest_conn.cc]: should be `wr->put_obj_init(` here?

Comment (yehudasa) [inline src/rgw/rgw_rest_conn.cc]: @cbodley oops, fixed

**Comment by ZVampirEM77** [comment]
@yehudasa Can you review this #16127 ? I think it is also a bug while copying object to remote dest

**Comment by yuriw** [comment]
http://pulpito.front.sepia.ceph.com:80/yuriw-2017-07-17_19:46:42-rgw-wip-yuri-testing2_2017_7_18-distro-basic-smithi/

**Pull request merged**